src/main/resources/application.yaml

```diff
@@ -1,4 +1,7 @@
 quarkus:
+  shutdown:
+    #  This Duration should be set enough for all sidecar requests(sync and async) completion
+    timeout: 5m
```

Comment:
Confused for this PR. Does this mean we will let the sidecar folo sealing action depends on the quarkus shutdown timeout? If so I don't think it is trustable, because we never knows the exact time for this configurations. Instead we should use lifecycle hooks: https://quarkus.io/guides/lifecycle

Reply:
@ligangty, this might be treated as a switch for graceful shutdown, we might not need to know the exact time for a build, the app will shutdown just after all the requests are completed regardless of the timeout, as long as the timeout we set is far greater than the time of normal build. Like 10h, 1d or so.
For your suggested lifecycle, I didn't find the proper way to do graceful shutdown, call asyncExit? I just did some trials and it seems the calling will initiate the Quarkus shutdown process, and then immediately return. Do you have any thoughts on this?

Reply:
I just update the timeout to 10h in yml example, and we could let PNC know the meaning of this setting if we use this method.

Reply:
Um...What I want to say is use this stop hook (https://quarkus.io/guides/lifecycle#listening-for-startup-and-shutdown-events) to do some post actions to make the whole tracking records submit like what we want to do before.
If I understand correctly, this graceful shutdown will guarantee all async http requests finished successfully? But I'm still not sure how this graceful shut down can make the seal action happen after all these requests completion here.

Reply:
@ligangty https://cn.quarkus.io/guides/lifecycle#graceful-shutdown, the graceful shutdown could ensure all running requests have completed before shutdown, I tested and it should be workable on async quarkus vert.x event. When shutdown is requesting, service will wait for the message event consuming completion then execute the shutdown. This might be the simplest way to guarantee.
I understand the stop hook for listening shutdown event you mentioned, one thing we may need to notice, for the lost dependency data mentioned in https://issues.redhat.com/browse/MMENG-3848, it seems that some lost dependencies also include the proxy folo ones(I'm not sure why it happens on 2.x), so this issue might not only be related to local downloads folo record, also need to take care the Indy proxy downloads folo record, so we need to guarantee all kind of download requests(proxy and local) are executed before shutdown.

Reply:
If the seal action will not happen in indy-sidecar, then this sounds ok to me. Feel free to merge and test.

Reply:
@ligangty sidecar will send PUT seal request to tracking service, tracking service does the seal action, is that your concern?

Reply:
Yeah. I don't remember the seal action is requested by which side, sidecar or PNC repo-driver? I it is the later one, the concern will gone because the sidecar has been shut down then. But it it is the former one, this will bring side-effect.

Reply:
@ligangty Yeah, I confirmed with Jan on this point that whether PNC seal report is definitely after the builder pod is dropped, he replied yes and the seal is done after cleaning the environment. For sidecar it will just request to save the folo record for each download in db, and then seal should be called after the builder pod is cleaned up.